### DISCUSSION TITLE
Cucumber JSON files are not matching with the cucumber-json-schema

### DISCUSSION THREAD

**Discussion by ramakkonatam** [webdriverio/webdriverio#11902] [Community]
Framework - Webdriverio - Cucumber - Typescript
Reporter - Wdio-cucumber-Json-reporter
CI - Jenkins
Integration tools - Zephyr

The issue we are encountering pertains to the generated JSON files from our test executions. Our objective is to upload the JSON results into the Zephyr tool from Jenkins using either the Zephyr plugin or the Zephyr API. Unfortunately, in both scenarios, we encounter a JSON parse error.

To address this challenge, we have thoroughly examined the Cucumber JSON files generated under the framework against the official JSON schema provided by Cucumber (https://github.com/cucumber/cucumber-json-schema/blob/main/schema.json). Regrettably, the JSON files do not align with the Cucumber JSON schema. Consequently, the Zephyr API, which validates the uploaded JSON files against the same schema, triggers a JSON parse error.

Now, the question arises: How can we ensure that the generated JSON files conform to the Cucumber JSON Schema? Is there a possibility to customize the JSON files to meet the required specifications? Additionally, we are puzzled as to why the WebDriverIO-Cucumber-generated JSON files fail to match the JSON schema (https://github.com/cucumber/cucumber-json-schema/blob/main/schema.json).

We are seeking guidance on aligning the JSON files with the Cucumber JSON schema and understanding whether there is flexibility to customize the JSON files. Furthermore, we would appreciate insights into why the JSON files generated by WebDriverIO-Cucumber do not adhere to the specified JSON schema (https://github.com/cucumber/cucumber-json-schema/blob/main/schema.json).
[package.json](https://github.com/webdriverio/webdriverio/files/13711593/package.json)
<img width="1479" alt="Screenshot 2023-12-19 at 10 46 13 AM" src="https://github.com/webdriverio/webdriverio/assets/143987445/72e9518b-9bdf-4323-a2e9-7693135174d9">
<img width="1512" alt="Screenshot 2023-12-19 at 10 48 48 AM" src="https://github.com/webdriverio/webdriverio/assets/143987445/507d8f14-8c2b-4216-a11f-1c34081b03df">
<img width="1512" alt="Screenshot 2023-12-19 at 10 49 03 AM" src="https://github.com/webdriverio/webdriverio/assets/143987445/7751d9b4-f88d-423e-b291-00386b52973e">
[schema.json](https://github.com/webdriverio/webdriverio/files/13711597/schema.json)

**Comment by ramakkonatam** [comment]
[FrameworkGenerated.json](https://github.com/webdriverio/webdriverio/files/13711608/FrameworkGenerated.json)
Generated json file attached

**Comment by christian-bromann** [comment]
Duplicate of https://github.com/webdriverio-community/wdio-cucumberjs-json-reporter/issues/396
